src/callbacks/steady_state.jl

```diff
@@ -0,0 +1,175 @@
+"""
+    SteadyStateCallback(; interval::Integer=0, dt=0.0, interval_size::Integer=10,
```

Comment:
ConvergenceConditionCallBack
ConvergenceTerminationCallBack
TerminationConditionCallBack

Reply:
`SteadyStateDetector`
`SteadyStateTerminator`

Reply:
As pointed out by @gregorgassner, "convergence" might be misleading. I also think the name should end with "callback".
So we have:
- `SteadyStateCallback`
- `TerminationConditionCallback`
- `SteadyStateReachedCallback` (by @gregorgassner)

`TerminationConditionCallback` is more general, but the callback is not. This could be a good name if we want to make the callback more general in the future.
`SteadyStateReachedCallback` makes the current use case very clear.
As I said before, I'm not a fan of `SteadyStateCallback`.

What do you think, @LasNikas , @svchb ?

Reply:
`SteadyStateReachedCallback` sounds good for me